lib/storage/data/external/GCP/GcpApis/mpuHelper.js

```
@@ -187,6 +187,10 @@ class MpuHelper {
                 };
                 this.service.deleteObject(delParams, err => {
                     if (err) {
+                        if (err.name === 'NoSuchKey') {
```

why do we need to _ignore_ this error in here?
the behavior of the API did not change, so we should not start to ignore the error?

---

Ignoring NoSuchKey here is about matching S3 semantics and keeping the cleanup idempotent, not relaxing validation in the caller-visible API. removeParts is a best-effort garbage-collection step running after we’ve already returned the composed object; when two retrying threads hit the same MPU bucket, one can delete a part before the other, and GC service responds with NoSuchKey (unlike S3, which treats the delete as a success). Letting that bubble up aborts the whole cleanup even though there is nothing left to delete. By swallowing just that code we align with S3’s “deleting a non-existent key succeeds” behaviour and avoid spurious failures caused by eventual-consistency/listed-but-already-removed parts. Any other error still makes us bail out, so external behaviour stays the same while retry/idempotency improves.